redis-utils/src/main/java/org/ballerinalang/redis/actions/AbstractRedisAction.java

```diff
@@ -1576,29 +1580,33 @@ private BValueArray createBStringArrayFromList(List<String> list) {
         return bMap;
     }
 
-    protected String[] createArrayFromBStringArray(BValueArray bStringArray) {
-        String[] stringArray = new String[(int) bStringArray.size()];
+    static String[] createStringArrayFromBArray(BArray bStringArray) {
+        String[] stringArray = new String[bStringArray.size()];
         for (int i = 0; i < bStringArray.size(); i++) {
             stringArray[i] = bStringArray.getString(i);
         }
         return stringArray;
     }
 
-    protected Map<String, String> createMapFromBMap(BMap<String, BString> bMap) {
-        Map<String, String> map = new HashMap<>();
-        bMap.getMap().keySet().forEach(item -> map.put(item, bMap.get(item).stringValue()));
+    static Map<String, Object> createMapFromBMap(BMap<String, Object> bMap) {
+        Map<String, Object> map = new LinkedHashMap<>();
+        for (Map.Entry<String, Object> entry : bMap.entrySet()) {
+            map.put(entry.getKey(), entry.getValue());
+        }
         return map;
     }
 
-    protected void setNullableReturnValues(BValue result, Context context) {
-        if (result == null) {
-            context.setReturnValues();
-        } else {
-            context.setReturnValues(result);
+    static BArray createBstringArrayFromBMap(BMap<String, BString> bMap) {
+        String[] keys = bMap.getKeys();
+        BArray bStringArray = new ArrayValueImpl(new String[bMap.size()]);
+        for (int i = 0; i < keys.length; i++) {
+            bStringArray.add(i, bMap.get(keys[i]).stringValue());
+            i++;
```

Comment:
Why do we need to increase the counter variable again inside a for loop?

Can we replace this with a for each loop?

Reply:
Updated in 8157b15